plasma_framework/contracts/src/framework/BlockController.sol

```diff
@@ -0,0 +1,76 @@
+pragma solidity ^0.5.0;
+
+import "./models/BlockModel.sol";
+import "./registries/VaultRegistry.sol";
+import "./utils/Operated.sol";
+
+contract BlockController is Operated, VaultRegistry {
+    uint256 private _childBlockInterval;
+    uint256 private _nextChildBlock;
+    uint256 private _nextDepositBlock;
+
+    mapping (uint256 => BlockModel.Block) private _blocks;
+
+    event BlockSubmitted(
+        uint256 blockNumber
```

Comment:
I think you could add `root hash` field to this event.

Reply:
Sound good idea to me, but I would like to keep the event to be same as existing one first. We could change after our first round of migration with elixir code.